### PR TITLE
Add some instructions to contrain EoS

### PR DESCRIPTION
**Pull request by weizmannk**

### CODE REVIEW

File: doc/constrain_R14_trend.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+## Contrain neuton stars Equation of State (EoS) 
+
+### create an output 
+
+First of all, you need to creat an output directory, this output will host all the data that will be used to constrain the EoS.
+    
+	mkdir -p ./nmma/output
+
+
+### gennerate EoS injection.json
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:10]: generate

File: doc/constrain_R14_trend.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+## Contrain neuton stars Equation of State (EoS) 
+
+### create an output 
+
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:4]: note that this assumes you are starting outside the nmma repository.

File: doc/constrain_R14_trend.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    
+	mkdir -p ./nmma/output
+
+
+### gennerate EoS injection.json
+
+Run this command line will genarate  a json file (injection.json)  with the BILBY processing of compact binary merging events (BNS, NSBH, optional), here it's BNS. This injection contents a simulation of the parameters : luminosity_distance, log10_mej_wind, KNphi, inclination_EM, KNtimeshift, geocent_time.
+At the end of the run, when the process is ok,  an injection.json will be find at  ./nmma/output directory.
+
+	nmma_create_injection --prior-file ./nmma/example_files/prior/ZTF_kn.prior --eos-file ./nmma/example_files/eos/ALF2.dat --binary-type BNS -f ./nmma/output/injection --n-injection 100 --original-parameters  --extension json
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:15]: Let's use the prior files in priors:

nmma_create_injection --prior-file ./nmma/priors/Bu2019lm.prior --eos-file ./nmma/example_files/eos/ALF2.dat --binary-type BNS -f ./nmma/output/injection --n-injection 100 --original-parameters  --extension json

File: doc/constrain_R14_trend.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	mkdir -p ./nmma/output/EMdata
+
+### lightcurve posterior 
+
+The next command line provide some EM posterios in this example we have a simulation of a sample of 100 events and and 26 of them are detectable by the ZTF. 
+
+	for macroeventID in {0..99}
+
+	do
+	 light_curve_analysis --model Bu2019lm --svd-path ./nmma/svdmodels --gptype tensorflow  --outdir ./nmma/output/EMdata/outdir/$macroeventID --label injection_Bu2019lm --prior ./nmma/example_files/prior/ZTF_kn.prior --tmin 0 --tmax 7 --dt 0.5 --error-budget 1.0 --nlive 256 --Ebv-max 0 --injection ./nmma/output/injection.json --injection-num $macroeventID --injection-detection-limit 22,22,22 --injection-outfile ./nmma/example_files/csv_lightcurve/outdir/$macroeventID/lc.csv --generation-seed 42 --filters g,r,i  --ztf-sampling --ztf-uncertainties --plot --remove-nondetections --optimal-augmentation --optimal-augmentation-filters u,g,r,i,z,y,J,H,K --optimal-augmentation-N 100
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:31]: I think should be:
mkdir ./nmma/output/EMdata/$macroeventID/
light_curve_analysis --model Bu2019lm --svd-path ./nmma/svdmodels --gptype tensorflow  --outdir ./nmma/output/EMdata/outdir/$macroeventID --label injection_Bu2019lm --prior ./nmma/priors/Bu2019lm.prior --tmin 0 --tmax 7 --dt 0.5 --error-budget 1.0 --nlive 256 --Ebv-max 0 --injection ./nmma/output/injection.json --injection-num $macroeventID --injection-detection-limit 22,22,22 --injection-outfile ./nmma/output/EMdata/$macroeventID/lc.csv --generation-seed 42 --filters g,r,i  --ztf-sampling --ztf-uncertainties --plot --remove-nondetections --optimal-augmentation --optimal-augmentation-filters u,g,r,i,z,y,J,H,K --optimal-augmentation-N 100

File: doc/constrain_R14_trend.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+### EoS from GW + EM 
+
+This allows to generate the EoS by combining the data of the events (EMdata) and those coming from the gravitational (GWdata).The 26 detectable events are {0,  3,  5,  7,  8, 10, 12, 13, 14, 15, 17, 19, 21, 22, 23, 24, 26, 27, 28, 31, 32,34, 36, 37, 38, 39}.
+A GW_EMdata files includind the combination of  GW and EM EoS data can be get at the direction of ./nmma/output . 
+
+	for macroeventID in {0,  3,  5,  7,  8, 10, 12, 13, 14, 15, 17, 19, 21, 22, 23, 24, 26, 27, 28, 31, 32,34, 36, 37, 38, 39}
+
+	do
+
+	 gwem_resampling --outdir ./nmma/output/GW_EMdata/$macroeventID --EMsamples ./nmma/output/EMdata/outdir/$macroeventID/injection_Bu2019lm_posterior_samples.dat --GWsamples  ./nmma/output/GWdata/outdir/inj_PhD_posterior_samples_$macroeventID.dat --EOS ./nmma/example_files/eos_sorted --nlive 8192 --GWprior ./nmma/priors/aligned_spin.priors --EMprior ./nmma/priors/EM.prior --total-ejecta-mass --Neos 5000
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:46]: How should we package the GWdata? What about the EOSs?

File: doc/constrain_R14_trend.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+	done
+
+
+### After all that plot EOS
+
+There are two pythons files (R14_trend_generate.py, R14_trend_plot.py)  on ./nmma/examples_files, which can use to visualize the EOS data.
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:53]: Can we add these to a directory called post-processing?

**Review by mcoughlin — CHANGES_REQUESTED** [review]
Looks good. I think there are a few bugs and we also need a way for folks to get access to the GW posteriors.

File: doc/constrain_R14_trend.md (hunk 5 above)
Comment (weizmannk) [reply doc/constrain_R14_trend.md:46]: the EoS is not to weight , eos_sorted,  it's 13 MB, 
it's this files that I have added for the first pull request , 
but it's will better if we can add it to the GW data on Zenedo , I think

Comment (mcoughlin) [reply doc/constrain_R14_trend.md:46]: It looks like Tim already has them on his github, so we can just point there for folks to download.

File: doc/constrain_R14_trend.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
+## Contrain neutron stars Equation of State (EoS) 
+
+### create an output 
+
+First of all, you need to creat an output directory, this output will host all the data that will be used to constrain the EoS.
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:5]: create

File: doc/constrain_R14_trend.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    
+	mkdir -p ./output
+
+
+### generate EoS injection.json
+
+Run this command line will genarate  a json file (injection.json)  with the BILBY processing of compact binary merging events (BNS, NSBH, optional), here it's BNS. This injection contents a simulation of the parameters : luminosity_distance, log10_mej_wind, KNphi, inclination_EM, KNtimeshift, geocent_time.
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:12]: Running

File: doc/constrain_R14_trend.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+	python R14_trend_generate.py
+
+Normaly when everything is ok you should be at are actully ./example_files/post-processing.
+You should obtain  GW_EM_R14trend_ZTF.dat  at  ../../output/Figures.
+Then please stay at the same location, that is mean here : ./example_files/post-processing and run the plot 
+
+	python R14_trend_plot.py
+
+Note : Sometimes when you use linux system or macos ... this erreur could be return
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:70]: Can we put this in the installation instructions instead?

File: doc/constrain_R14_trend.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+Then got to :
+	
+	cd ./example_files/post-processing
+	
+The first file to run is :
+
+	python R14_trend_generate.py
+
+Normaly when everything is ok you should be at are actully ./example_files/post-processing.
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:64]: Normally.

File: doc/constrain_R14_trend.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
+## Contrain neutron stars Equation of State (EoS) 
+
```
Comment (mcoughlin) [inline doc/constrain_R14_trend.md:2]: I would put the link to Zenodo and Tim's EOS repository and the discussion of which files to get up here.

**Review by mcoughlin — CHANGES_REQUESTED** [review]
Also looking good.

File: doc/constrain_R14_trend.md (hunk 9 above)
Comment (weizmannk) [reply doc/constrain_R14_trend.md:70]: Ok I will try to add an argument and test it before  change this command lines

File: doc/constrain_R14_trend.md (hunk 11 above)
Comment (mcoughlin) [reply doc/constrain_R14_trend.md:2]: I recall a few were decided to not be fully converged. Clarify with Peter

Comment (weizmannk) [reply doc/constrain_R14_trend.md:2]: ok

**Pull request merged**